nx584/controller.py

```diff
@@ -248,19 +248,19 @@ def send_nack(self):
     def get_zone_name(self, number):
         self._queue.append([0x23, number - 1])
 
-    def arm_stay(self, partition=1):
-        self._queue.append([0x3E, 0x00, partition])
+    def arm_stay(self, partition):
+        self._queue.append([0x3E, 0x00, int(partition)])
```

Comment:
Please don't just type-convert everywhere. This is internal, so the `int()` definitely doesn't belong here.